Completed Robot task

SiarheiMatsevich opened the pull request:
- To apply ScreenshotListener please use the following command:
 `robot --listener .\resources\ScreenshotListener.py tests\smoke.robot`

SiarheiMatsevich committed:
Completed Robot task:
- To apply ScreenshotListener please use the following command:
`robot --listener .\resources\ScreenshotListener.py tests\smoke.robot`

akhmialeuski commented:
Looks like something is broken

```
PS C:\Users\Anatol_Khmialeuski\PycharmProjects\ATMwPB_Siarhei Matsevich> python -m robot --listener .\resources\ScreenshotListener.py tests\smoke.robot
==============================================================================
Smoke :: Coverage for "Login" and "Add to cart" features.
==============================================================================

DevTools listening on ws://127.0.0.1:51661/devtools/browser/aae9baef-b3b2-4d45-b836-e2c8130d3a82
Login Happy Path :: Log in with registered user credentials.          | FAIL |
Parent suite setup failed:
ElementClickInterceptedException: Message: element click intercepted: Element <a class="nav-link" href="#" id="signin2" data-toggle="modal" data-target="#signInModal" style="display: block;">...</a> is not clickable at point (363, 86). Other element would receive the click: <img class="d-block img-fluid" src="Samsung1.jpg" alt="First slide">
  (Session info: chrome=123.0.6312.106)
Stacktrace:
        GetHandleVerifier [0x00007FF79CC47032+63090]
        (No symbol) [0x00007FF79CBB2C82]
        (No symbol) [0x00007FF79CA4EC65]
        (No symbol) [0x00007FF79CA9BB34]
        (No symbol) [0x00007FF79CA99954]
        (No symbol) [0x00007FF79CA97164]
        (No symbol) [0x00007FF79CA95EF9]
        (No symbol) [0x00007FF79CA8A708]
        (No symbol) [0x00007FF79CAB6FDA]
        (No symbol) [0x00007FF79CA8A00A]
        (No symbol) [0x00007FF79CAB71F0]
        (No symbol) [0x00007FF79CAD3412]
        (No symbol) [0x00007FF79CAB6D83]
        (No symbol) [0x00007FF79CA883A8]
        (No symbol) [0x00007FF79CA89441]
        GetHandleVerifier [0x00007FF79D0425AD+4238317]
        GetHandleVerifier [0x00007FF79D07F70D+4488525]
        GetHandleVerifier [0x00007FF79D0779EF+4456495]
        GetHandleVerifier [0x00007FF79CD20576+953270]
        (No symbol) [0x00007FF79CBBE54F]
        (No symbol) [0x00007FF79CBB9224]
        (No symbol) [0x00007FF79CBB935B]
        (No symbol) [0x00007FF79CBA9B94]
        (No symbol) [0x00007FF99F9B257D]
        RtlUserThreadStart [0x00007FF9A05EAA58+40]
------------------------------------------------------------------------------
Adding item to the cart :: Add item to cart. Basic scenario.          | FAIL |
Parent suite setup failed:
ElementClickInterceptedException: Message: element click intercepted: Element <a class="nav-link" href="#" id="signin2" data-toggle="modal" data-target="#signInModal" style="display: block;">...</a> is not clickable at point (363, 86). Other element would receive the click: <img class="d-block img-fluid" src="Samsung1.jpg" alt="First slide">
  (Session info: chrome=123.0.6312.106)
Stacktrace:
        GetHandleVerifier [0x00007FF79CC47032+63090]
        (No symbol) [0x00007FF79CBB2C82]
        (No symbol) [0x00007FF79CA4EC65]
        (No symbol) [0x00007FF79CA9BB34]
        (No symbol) [0x00007FF79CA99954]
        (No symbol) [0x00007FF79CA97164]
        (No symbol) [0x00007FF79CA95EF9]
        (No symbol) [0x00007FF79CA8A708]
        (No symbol) [0x00007FF79CAB6FDA]
        (No symbol) [0x00007FF79CA8A00A]
        (No symbol) [0x00007FF79CAB71F0]
        (No symbol) [0x00007FF79CAD3412]
        (No symbol) [0x00007FF79CAB6D83]
        (No symbol) [0x00007FF79CA883A8]
        (No symbol) [0x00007FF79CA89441]
        GetHandleVerifier [0x00007FF79D0425AD+4238317]
        GetHandleVerifier [0x00007FF79D07F70D+4488525]
        GetHandleVerifier [0x00007FF79D0779EF+4456495]
        GetHandleVerifier [0x00007FF79CD20576+953270]
        (No symbol) [0x00007FF79CBBE54F]
        (No symbol) [0x00007FF79CBB9224]
        (No symbol) [0x00007FF79CBB935B]
        (No symbol) [0x00007FF79CBA9B94]
        (No symbol) [0x00007FF99F9B257D]
        RtlUserThreadStart [0x00007FF9A05EAA58+40]
------------------------------------------------------------------------------
Smoke :: Coverage for "Login" and "Add to cart" features.             | FAIL |
Suite setup failed:
ElementClickInterceptedException: Message: element click intercepted: Element <a class="nav-link" href="#" id="signin2" data-toggle="modal" data-target="#signInModal" style="display: block;">...</a> is not clickable at point (363, 86). Other element would receive the click: <img class="d-block img-fluid" src="Samsung1.jpg" alt="First slide">
  (Session info: chrome=123.0.6312.106)
Stacktrace:
        GetHandleVerifier [0x00007FF79CC47032+63090]
        (No symbol) [0x00007FF79CBB2C82]
        (No symbol) [0x00007FF79CA4EC65]
        (No symbol) [0x00007FF79CA9BB34]
        (No symbol) [0x00007FF79CA99954]
        (No symbol) [0x00007FF79CA97164]
        (No symbol) [0x00007FF79CA95EF9]
        (No symbol) [0x00007FF79CA8A708]
        (No symbol) [0x00007FF79CAB6FDA]
        (No symbol) [0x00007FF79CA8A00A]
        (No symbol) [0x00007FF79CAB71F0]
        (No symbol) [0x00007FF79CAD3412]
        (No symbol) [0x00007FF79CAB6D83]
        (No symbol) [0x00007FF79CA883A8]
        (No symbol) [0x00007FF79CA89441]
        GetHandleVerifier [0x00007FF79D0425AD+4238317]
        GetHandleVerifier [0x00007FF79D07F70D+4488525]
        GetHandleVerifier [0x00007FF79D0779EF+4456495]
        GetHandleVerifier [0x00007FF79CD20576+953270]
        (No symbol) [0x00007FF79CBBE54F]
        (No symbol) [0x00007FF79CBB9224]
        (No symbol) [0x00007FF79CBB935B]
        (No symbol) [0x00007FF79CBA9B94]
        (No symbol) [0x00007FF99F9B257D]
        RtlUserThreadStart [0x00007FF9A05EAA58+40]


2 tests, 0 passed, 2 failed
==============================================================================
Output:  C:\Users\Anatol_Khmialeuski\PycharmProjects\ATMwPB_Siarhei Matsevich\output.xml
Log:     C:\Users\Anatol_Khmialeuski\PycharmProjects\ATMwPB_Siarhei Matsevich\log.html
Report:  C:\Users\Anatol_Khmialeuski\PycharmProjects\ATMwPB_Siarhei Matsevich\report.html

```

akhmialeuski commented:
![image](https://github.com/SiarheiMatsevich/python-automation-mentoring/assets/34238475/a0ee1e62-0f3c-4882-b545-b1810ce5782e)

akhmialeuski commented:
![image](https://github.com/SiarheiMatsevich/python-automation-mentoring/assets/34238475/3e4f633b-98d3-49a7-ac1f-36411d9f0834)
As you can see, there's no menu at the top for logging. When you open the page full screen, there are.

akhmialeuski reviewed (changes requested):
All tests failed

SiarheiMatsevich committed:
Completed Robot task:
- Browser window maximizing step was added to avoid a global navigation component displaying when viewport is less than 992px (The fix will only apply if screen with is more than 992px)

SiarheiMatsevich commented:
> All tests failed

I have not caught that on my local machine. Thanks!
Fixed  in d9b41d6ca3ec94a232a7626b289dc167fdaa9124

```
(venv) D:\PycharmProjects\python-automation-mentoring git:[task10]
robot --listener .\resources\ScreenshotListener.py tests\smoke.robot
==============================================================================
Smoke :: Coverage for "Login" and "Add to cart" features.                     
==============================================================================

DevTools listening on ws://127.0.0.1:65478/devtools/browser/59f88ab7-620e-4435-bc75-1e095704bd2b
Login Happy Path :: Log in with registered user credentials.          
DevTools listening on ws://127.0.0.1:65497/devtools/browser/00e87d05-e748-47c7-8266-c2a5602ef9a8
Login Happy Path :: Log in with registered user credentials.          | PASS |
------------------------------------------------------------------------------
Adding item to the cart :: Add item to cart. Basic scenario.          | PASS |
------------------------------------------------------------------------------
Smoke :: Coverage for "Login" and "Add to cart" features.             | PASS |
2 tests, 2 passed, 0 failed
==============================================================================
Output:  D:\PycharmProjects\python-automation-mentoring\output.xml
Log:     D:\PycharmProjects\python-automation-mentoring\log.html
Report:  D:\PycharmProjects\python-automation-mentoring\report.html
```

akhmialeuski reviewed (approved):
Done
